docs/docs/api/producer.md

```diff
@@ -10,15 +10,22 @@ import { Producer } from "@c11/engine.macro"
 
 ## Overview
 
-A `Producer` is a function that reacts to specific state changes and produces new data through specific state updates. It can also produce effects. 
+A `Producer` is simply a function that gets executed when something interesting
+(a trigger) changes in state.
```

Comment:
Also, note that producers don't necessarily need a trigger as they can work even without a trigger:
```
const foo: producer = ({ value: Update.foo.value }) => {
// interact with a service
value.set(newData)
}
```
This producer will be called only once, when the Engine starts or when the component on which it is declared is mounted.

But, it has it's niche cases in which it should be used - this is an advanced pattern and not something you should see early in development.

Reply:
> This producer will be called only once, when the Engine starts or when the component on which it is declared is mounted.

Do all producers of a component are always executed at least once every time it is mounted? If I have something like `{myBool && <MyComponent />}`, then every time `myBool` changes from `false` to `true`, will all the producers of `MyComponent` will execute?

Reply:
Yes, producers always execute at least once because they receive the state they are interested in, regardless what that state is. Producers that do not have an Observe can be considered a type of singleton - they can't be executed more than once if they are defined at the Engine level. At view level, they will be executed once per mounting.

The idea behind producers, and this is something we should clarify, is that they do synchronization, they are responsible with keeping stuff in sync. The execution part of a producer is its effort of keeping in sync an internal or external state. In other words, every time producers execute they will analyse the condition of the state and decide if they need to sync it or not. This is in stark contrast to the defacto in programming where computation is driven by direct commands (or even events where you're listening to asyc commands)

Because you don't have any control regarding calling a producer the only thing you can control is evaluating the state and deciding if its suitable or not.

That's why producers are always called. No one knows when a certain producer needs to do its execution, except the producer itself.

Reply:
However, the situation where a producer doesn't neither an Observe nor a Get, thus not relying on anything in regards of the state it's a dubious situation. Even if it should do a one off execution that should still be recorded that it happened somewhere - producer doesn't have control if its called once or twice - even at the root level.

Such producers are most likely an anti-pattern, if you can think of any situation where this would be appropriate, let me know.

It might be something regarding a view - producer relation where you would need to do something on an html element after the view mounting. In this situation you would check the dom and see if it needs modifying - like scrolling for example or listening to scroll and reacting to it. Although i think this behviour can be intermediated through the state as well.